auth/src/main/java/org/openedx/auth/presentation/signup/SignUpViewModel.kt

```diff
@@ -203,32 +215,44 @@ class SignUpViewModel(
     }
 
     fun socialAuth(fragment: Fragment, authType: AuthType) {
+        logEvent(AuthAnalyticsEvent.SOCIAL_REGISTER_CLICKED, buildMap {
+            put(AuthAnalyticsKey.METHOD.key, authType.methodName.lowercase())
+        })
         _uiState.update { it.copy(isLoading = true) }
         viewModelScope.launch {
             withContext(Dispatchers.IO) {
                 runCatching {
                     oAuthHelper.socialAuth(fragment, authType)
                 }
+            }.onSuccess { socialAuth ->
+                socialAuth.checkToken()
+            }.onFailure { exception ->
+                _uiState.update { it.copy(isLoading = false) }
+                logLogistrationFailureEvent(
+                    AuthAnalyticsEvent.SOCIAL_AUTH_FAILURE,
+                    authType,
+                    exception
+                )
             }
-                .getOrNull()
-                .checkToken()
         }
     }
 
-    private suspend fun SocialAuthResponse?.checkToken() {
-        this?.accessToken?.let { token ->
+
+    private suspend fun SocialAuthResponse.checkToken() {
+        accessToken.let { token ->
             if (token.isNotEmpty()) {
                 exchangeToken(this)
             } else {
                 _uiState.update { it.copy(isLoading = false) }
             }
-        } ?: _uiState.update { it.copy(isLoading = false) }
+        }
     }
 
     private suspend fun exchangeToken(socialAuth: SocialAuthResponse) {
         runCatching {
             interactor.loginSocial(socialAuth.accessToken, socialAuth.authType)
         }.onFailure {
+            logLogistrationFailureEvent(AuthAnalyticsEvent.SIGN_IN_FAILURE, socialAuth.authType, it)
```

Comment:
We don't need this call because we won' be able to differentiate between this and actual sign in call.

Reply:
Nice find!